src/modules/common/components/SocialShare/SocialShare.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import { t } from 'modules/i18n/utils/intl';
+import { Menu, MenuItem, MenuList } from '@material-ui/core';
+import {
+  FacebookShareButton,
+  TelegramShareButton,
+  TwitterShareButton,
+} from 'react-share';
+import { TwitterIcon } from 'modules/common/components/Icons/TwitterIcon';
+import { FacebookIcon } from 'modules/common/components/Icons/FacebookIcon';
+import { TelegramIcon } from 'modules/common/components/Icons/TelegramIcon';
+import { useSocialShareStyles } from './useSocialShareStyles';
+
+interface ISocialShareProps {
+  buttonContent: JSX.Element;
+  url?: string;
+  titleString?: string;
+  description?: string;
+}
+
+const currentUrl = window.location.href;
+
+export const SocialShare = ({
+  buttonContent,
+  url = currentUrl,
+  titleString,
+  description,
+}: ISocialShareProps) => {
+  const classes = useSocialShareStyles();
+
+  const [anchorEl, setAnchorEl] = useState(null);
+
+  const handleClick = (event: any) => {
```

Comment:
I don't know what type is needed, everything I've tried doesn't work...

Reply:
Try this

```
  const [anchorEl, setAnchorEl] = useState<HTMLElement>();

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(undefined);
  };

```

Reply:
nope